packages/react/src/SelectPanel/SelectPanel.features.stories.tsx

```diff
@@ -369,3 +369,70 @@ export const WithGroups = () => {
     />
   )
 }
+
+export const ModalVariant = () => {
+  const initialItems = [items[0], items[1]]
+  const [selected, setSelected] = React.useState<ItemInput[]>(initialItems)
+  const [filter, setFilter] = React.useState('')
+  const filteredItems = items.filter(item => item.text.toLowerCase().startsWith(filter.toLowerCase()))
+  const [open, setOpen] = useState(false)
+  const buttonRef = useRef<HTMLButtonElement>(null)
+
+  return (
+    <>
+      <h1>Multi Select Panel as Modal</h1>
+      <SelectPanel
+        variant="modal"
+        onCancel={() => setSelected(initialItems)}
+        renderAnchor={({children, 'aria-labelledby': ariaLabelledBy, ...anchorProps}) => (
+          <Button aria-labelledby={` ${ariaLabelledBy}`} {...anchorProps}>
+            {children ?? 'Select Labels'}
+          </Button>
+        )}
+        anchorRef={buttonRef}
+        placeholderText="Filter Labels"
+        open={open}
+        onOpenChange={setOpen}
+        items={filteredItems}
+        selected={selected}
+        onSelectedChange={setSelected}
+        onFilterChange={setFilter}
+      />
+    </>
+  )
+}
+
+export const ModalVariantWithSecondaryAction = () => {
+  const initialItems = [items[0], items[1]]
+  const [selected, setSelected] = React.useState<ItemInput[]>(initialItems)
+  const [filter, setFilter] = React.useState('')
+  const filteredItems = items.filter(item => item.text.toLowerCase().startsWith(filter.toLowerCase()))
+  const [open, setOpen] = useState(false)
+  const buttonRef = useRef<HTMLButtonElement>(null)
+
+  return (
+    <>
+      <h1>Multi Select Panel as Modal with secondary action</h1>
+      <SelectPanel
+        variant="modal"
+        onCancel={() => setSelected(initialItems)}
+        // backward compatible API choice
+        // TODO: improve this API, rename it to a single secondaryAction or secondaryFooterSlotSomething
+        footer={<Button size="small">Edit labels</Button>}
```

Comment:
Todo 1: While footer is backward compatible, it's not obvious what it means. What we want here is a "secondary action".

Looking at usage ([3 instances](https://primer-query.githubapp.com/?name=selectpanel&attribute=footer)), we have
- Button
- LinkButton
- custom FooterActions component that has a spinner or an error message! (these cases are filling in for missing features and we should be able to simplify them with error messages and loading states)